spirv_glsl.cpp

```diff
@@ -10239,9 +10244,13 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 		// If an expression is mutable and forwardable, we speculate that it is immutable.
 		AccessChainMeta meta;
 		bool ptr_chain = opcode == OpPtrAccessChain;
-		auto e = access_chain(ops[2], &ops[3], length - 3, get<SPIRType>(ops[0]), &meta, ptr_chain);
+		auto target_type = get<SPIRType>(ops[0]);
+		auto e = access_chain(ops[2], &ops[3], length - 3, target_type, &meta, ptr_chain);
 
-		auto &expr = set<SPIRExpression>(ops[1], move(e), ops[0], should_forward(ops[2]));
+		auto &expr = (!backend.can_declare_struct_inline && flattened_buffer_blocks.count(ops[2]) &&
```

Comment:
Make the boolean condition here into a temporary. Also add a comment about the rationale, that we might not be able to build composites inline on some backends, and they have to be bound to a temporary.